File: util/android/src/main/java/com/dmi/util/base/BaseUtils.java
package com.dmi.util.base;

import java.util.List;
import java.util.ListIterator;

import java8.util.function.Predicate;
import me.tatarka.simplefragment.SimpleFragment;
import me.tatarka.simplefragment.SimpleFragmentManagerProvider;

abstract class BaseUtils {
    public static boolean handleAttachedFragments(SimpleFragmentManagerProvider managerProvider,
                                                  Predicate<BaseFragment> handler) {
        List<SimpleFragment> fragments = managerProvider.getSimpleFragmentManager().getAttachedFragments();
        ListIterator<SimpleFragment> iterator = fragments.listIterator(fragments.size());
        while (iterator.hasPrevious()) {
            SimpleFragment fragment = iterator.previous();
            if (fragment instanceof BaseFragment) {
                if (handler.test((BaseFragment) fragment)) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: util/android/src/main/kotlin/com/dmi/util/base/BaseUtils.kt
package com.dmi.util.base

import me.tatarka.simplefragment.SimpleFragmentManagerProvider

internal object BaseUtils {
    fun handleAttachedFragments(managerProvider: SimpleFragmentManagerProvider,
                                handle: (BaseFragment) -> Boolean): Boolean {
        val fragments = managerProvider.simpleFragmentManager.attachedFragments
        val iterator = fragments.listIterator(fragments.size)
        while (iterator.hasPrevious()) {
            val fragment = iterator.previous()
            if (fragment is BaseFragment) {
                if (handle(fragment)) {
                    return true
                }
            }
        }
        return false
    }
}
